src/reformatters/common/region_job.py

```diff
@@ -162,7 +167,7 @@ def read_data(
         self,
         coord: SOURCE_FILE_COORD,
         data_var: DATA_VAR,
-    ) -> ArrayFloat32:
+    ) -> ArrayFloat32 | ArrayInt16:
```

Comment:
Maybe a bit weird to add these as needed, perhaps we should hint with a more generic type

Reply:
yeah unless we got super fancy and somehow pull out the element type from the DATA_VAR (I don't think we should try), lets do

```suggestion
    ) -> np.ndarray:
```

Reply:
Type checker is unhappy, with just `np.ndarray`, we'd need to do something like `np.ndarray[Any, Any]` .

I've defined
```
type ArrayND[D: np.generic] = np.ndarray[tuple[int, ...], np.dtype[D]]
```

and the common `read_data` return type looks like:

```
    def read_data(
        self,
        coord: SOURCE_FILE_COORD,
        data_var: DATA_VAR,
    ) -> ArrayND[np.generic]:
 ```

`mypy` seems happy, so I'll run with this but lmk if there might be problems with this